src/modules/material-requests/controllers/material-requests.controller.ts

```diff
@@ -154,29 +159,53 @@ export class MaterialRequestsController {
 		dto.isPending = true;
 		dto.size = 100;
 
-		const materialRequests = await this.materialRequestsService.findAll(dto, {
-			userProfileId: user.getId(),
-			userGroup: user.getGroupId(),
-			isPersonal: true,
-		});
-
-		materialRequests.items.forEach(
-			(materialRequest: MaterialRequest) =>
-				(materialRequest.contactMail = materialRequest.contactMail.find(
-					(contact) => contact.contact_type === 'primary'
-				)?.email)
-		);
-
-		await this.materialRequestsService.sendRequestList(
-			materialRequests.items,
-			sendRequestListDto,
-			{
-				firstName: user.getFirstName(),
-				lastName: user.getLastName(),
-			}
-		);
-
-		return { message: 'success' };
+		try {
+			const materialRequests = await this.materialRequestsService.findAll(dto, {
+				userProfileId: user.getId(),
+				userGroup: user.getGroupId(),
+				isPersonal: true,
+			});
+
+			materialRequests.items.forEach(
+				(materialRequest: MaterialRequest) =>
+					(materialRequest.contactMail = materialRequest.contactMail.find(
+						(contact) => contact.contact_type === 'primary'
+					)?.email)
+			);
+
+			await this.materialRequestsService.sendRequestList(
+				materialRequests.items,
+				sendRequestListDto,
+				{
+					firstName: user.getFirstName(),
+					lastName: user.getLastName(),
+				}
+			);
+
+			await Promise.all(
+				materialRequests.items.map(async (materialRequest: MaterialRequest) => {
+					await this.materialRequestsService.updateMaterialRequest(
+						materialRequest.id,
+						user.getId(),
+						{
+							type: materialRequest.type as Lookup_App_Material_Request_Type_Enum,
+							reason: materialRequest.reason,
+							organisation: materialRequest.organisation,
+							requester_capacity:
+								materialRequest.requesterCapacity as Lookup_App_Material_Request_Requester_Capacity_Enum,
```

Comment:
Shouldn't the MaterialRequest interface already contain:
```
export interface MaterialRequest {
	id: string;
	objectSchemaIdentifier: string;
	objectSchemaName: string;
	objectMeemooIdentifier: string;
	objectMeemooLocalId?: string;
	objectDctermsFormat: MediaFormat;
	objectThumbnailUrl: string;
	profileId: string;
	reason: string;
	createdAt: string;
	updatedAt: string;
	type: string;
	isPending: boolean;
	requesterId: string;
	requesterFullName: string;
	requesterMail: string;
        requesterCapacity: Lookup_App_Material_Request_Requester_Capacity_Enum
	maintainerId: string;
	maintainerName: string;
	maintainerSlug: string;
	requesterUserGroupId?: string;
	requesterUserGroupName?: string;
	requesterUserGroupLabel?: string;
	requesterUserGroupDescription?: string;
	maintainerLogo?: string;
	contactMail?: any;
	organisation?: string | null;
}
```
So we don't need to cast this here anymore.